jormungandr/src/settings/start/mod.rs

```diff
@@ -264,44 +269,40 @@ fn generate_network(
             .collect()
     });
 
-    let mut profile = poldercast::NodeProfileBuilder::new();
-
-    if let Some(address) = command_arguments.public_address.clone() {
-        profile.address(address);
-    } else if let Some(address) = p2p.public_address {
-        profile.address(address);
-    }
-
-    let legacy_node_id = p2p
-        .public_id
-        .unwrap_or_else(|| poldercast::Id::generate(rand::thread_rng()));
-    profile.id(legacy_node_id);
-
-    for (topic, interest_level) in p2p
-        .topics_of_interest
-        .unwrap_or_else(config::default_interests)
-    {
-        let sub = poldercast::Subscription {
-            topic: topic.0,
-            interest: interest_level.0,
-        };
-        profile.add_subscription(sub);
-    }
+    // TODO: do we want to check that we end up with a valid address?
+    // Is it possible for a node to specify no public address?
```

Comment:
Should be possible, it's a non-public node then which should also not gossip about itself.